src/deluge/playback/playback_handler.cpp

```diff
@@ -2521,18 +2521,18 @@ bool PlaybackHandler::tryGlobalMIDICommandsOff(MIDIDevice* device, int32_t chann
 	return foundAnything;
 }
 
-void PlaybackHandler::programChangeReceived(int32_t channel, int32_t program) {
-
-	/*
-    // If user assigning MIDI commands, do that
-    if (subMode == UI_MODE_MIDI_LEARN) {
-    	if (getCurrentUI() == &soundEditor && soundEditor.noteOnReceivedForMidiLearn(channel + 16, program, 127)) {}
-    	return;
-    }
-
-
-    tryGlobalMIDICommands(channel + 16, program);
-    */
+void PlaybackHandler::programChangeReceived(MIDIDevice* fromDevice, int32_t channel, int32_t program) {
+	// If user assigning MIDI commands, do that
+	if (currentUIMode == UI_MODE_MIDI_LEARN) {
+		if (getCurrentUI()->pcReceivedForMidiLearn(fromDevice, channel, program)) {}
```

Comment:
I suggest we just don't do this, and only allow PC learning for non-momentary global commands

Reply:
This is the real code path being used, and it seems to be the most straightforward path to implement thi  functionality while minimizing the cognitive load required to reason about signaling paths for other  devs in the future. 

Its the precise same path used for learning notes for commands. Right here: https://github.com/SynthstromAudible/DelugeFirmware/pull/801/files/2786b7007060d69fd8ca9672b4ef9b99c9033df3#diff-17a7ec38a9275b782f61ac4d14a3d043beedc0eb78fab024ea73802059ba76e9L2538


I did my best to make it as consistent as possible with the existing logic for notes. Not because i felt that there would be other places i could imagine where we use program changes for some other purpose, but just because it felt like the most maintainable / symetrical path. 

This approach _should_ not imply for example:
- Anything and everything can be learned by PC. PC is only learnable by views that  override `pcReceivedForMidiLearn` method. Which in this case, is only `SoundEditor`? 

Since its line 2527 that passes the control of the implemneted functionality into the soundeditor, line 2529 is therefore not used and only exists to not deviate from the pre-established pattern already used for note on messages further down.

Reply:
Global commands are learnt exclusively in the soundEditor - just call that using the same logic as the midiccReceived function

Reply:
we've eliminated most of the unnecessary code paths here but this logic remains necessary to maintain consistency and to actual trigger the sound editor's midi learn stack which arrives by the same branching logic for notes, but carry a far more specific use case. Still, the consistency in order of objects having the responsibility for this class of message has deviated as  little as possible, sparing only code paths that were not going to be used for this feature.